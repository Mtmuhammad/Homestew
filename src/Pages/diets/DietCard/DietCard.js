import React from "react";
import "./DietCard.scss";

const DietCard = ({ title, img, text }) => {
  return (
    <div className="col-12 col-sm-6 col-lg-4 mb-5">
      <article className="text-center diet-post">
        <div className="diet-img">
          <img src={img} alt={title} className="img-fluid" />
        </div>
        <div className="diet-content">
          <h2>{title}</h2>
          <p className="px-5 px-md-3">{text}</p>
        </div>
      </article>
    </div>
  );
};

export default DietCard;
