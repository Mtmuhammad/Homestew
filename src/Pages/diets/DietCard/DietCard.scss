@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.diet {
  & .fa-utensils {
    color: $tertiary-color;
    font-size: 3.8rem;
    margin-left: 1rem;
  }

  & img {
    height: 50rem !important;
    border-radius: 5px;

    @include respond(phone) {
      height: 40rem !important;
    }
    @include respond(sphone) {
      height: 30rem !important;
    }
  }

  &-post {
    margin-bottom: 5rem;
  }

  &-content {
    text-align: center;
    & h2 {
      font-size: 3.5rem;
      letter-spacing: 1px;
      margin: 2rem 0;
      color: $primary-green;
    }

    & p {
      line-height: 1.7;
      font-weight: 400;
      font-size: 1.8rem;
    }
  }
}
