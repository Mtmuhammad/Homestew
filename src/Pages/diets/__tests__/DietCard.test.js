import React from "react";
import { render, screen } from "@testing-library/react";
import DietCard from "../DietCard/DietCard";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<DietCard />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<DietCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render img", () => {
  render(
    <DietCard
      title={"Ketogenic"}
      text={"this is a test text"}
      img={"ketogenic.jpg"}
    />
  );

  const img = screen.getByAltText(/ketogenic/i);
  expect(img).toBeInTheDocument();
  expect(img).toHaveClass("img-fluid");
});

test("should render heading", () => {
  render(
    <DietCard
      title={"Ketogenic"}
      text={"this is a test text"}
      img={"ketogenic.jpg"}
    />
  );
  const heading = screen.getByRole("heading", { name: /ketogenic/i });
  expect(heading).toBeInTheDocument();
});
