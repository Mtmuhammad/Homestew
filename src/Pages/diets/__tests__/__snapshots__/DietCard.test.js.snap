// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-12 col-sm-6 col-lg-4 mb-5"
  >
    <article
      class="text-center diet-post"
    >
      <div
        class="diet-img"
      >
        <img
          class="img-fluid"
        />
      </div>
      <div
        class="diet-content"
      >
        <h2 />
        <p
          class="px-5 px-md-3"
        />
      </div>
    </article>
  </div>
</DocumentFragment>
`;
